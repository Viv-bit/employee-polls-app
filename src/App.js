import React, { useEffect } from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import LoadingBar from "react-redux-loading";
import { handleInitialData } from "./redux/actions/shared";

// Import all components here
import Login from "./views/Login/Login";
import Dashboard from "./views/Dashboard/Dashboard";
import AddQuestion from "./views/AddQuestion/index";
import LeaderBoard from "./views/Leaderboard/Leaderboard";
import NotFound from "./views/NotFound";
import Switch from "./views/Switch";
import AuthRoute from "./routes/AuthRoute";
import Navigation from "./components/Navigation";

const App = () => {
  const dispatch = useDispatch();
  const authenticatedUser = useSelector(
    ({ authenticatedUser }) => authenticatedUser
  );

  useEffect(() => {
    dispatch(handleInitialData());
  }, [dispatch]);

  return (
    <Router>
      <div className="App" data-testid="app">
        <LoadingBar />
        {authenticatedUser && <Navigation />}
        <Routes>
          <Route path="/" element={<Login />} />
          <Route
            path="/home"
            element={
              <AuthRoute>
                {" "}
                <Dashboard />
              </AuthRoute>
            }
          />
          <Route
            path="add"
            element={
              <AuthRoute>
                <AddQuestion />
              </AuthRoute>
            }
          />
          <Route
            path="questions/:question_id"
            element={
              <AuthRoute>
                <Switch />
              </AuthRoute>
            }
          />
          <Route
            path="leaderboard"
            element={
              <AuthRoute>
                <LeaderBoard />
              </AuthRoute>
            }
          />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </div>
    </Router>
  );
};

export default App;
