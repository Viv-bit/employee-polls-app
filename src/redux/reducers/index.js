import { combineReducers } from "redux";
import { loadingBarReducer as loadingBar } from "react-redux-loading";
import authenticatedUser from "./authUser";
import users from "./users";
import questions from "./questions";

export default combineReducers({
  authenticatedUser,
  users,
  questions,
  loadingBar,
});
