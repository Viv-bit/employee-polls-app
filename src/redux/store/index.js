import { createStore } from "redux";
import reducersRoot from "../reducers";
import middleware from "../middleware";

const reducersConfig = (state, action) => reducersRoot(state, action);

export const initCreateStore = () => createStore(reducersConfig, middleware);

const store = initCreateStore();

export default store;
