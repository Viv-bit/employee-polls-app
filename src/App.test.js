import { render, screen } from "@testing-library/react";
import App from "./App";
import { Provider } from "react-redux";
import { initCreateStore } from "./redux/store";
import renderer from "react-test-renderer";

describe("test for app component", () => {
  test("if app is mounted on DOM", () => {
    render(
      <Provider store={initCreateStore()}>
        <App />
      </Provider>
    );
    const divElement = screen.getByText(/app/i);

    expect(divElement).toBeDefined();
  });

  test("renders correctly", () => {
    const { asFragment } = render(
      <Provider store={initCreateStore()}>
        <App />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
