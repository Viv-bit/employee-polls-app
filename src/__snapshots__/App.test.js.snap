// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for app component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-testid="app"
  >
    <div>
      <div
        class=""
        style="opacity: 0; transform: translate3d(-100%, 0px, 0px); transition: transform 800ms linear 0s; width: 100%; will-change: transform, opacity; height: 3px; background-color: red; position: absolute;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <div
      role="authenticate"
    >
      <div
        class="container"
      >
        <div
          class="align-items-center justify-content-center min-vh-100 row"
        >
          <div
            class="col-md-6 col-12"
          >
            <div
              class="card"
              style="box-shadow: 0px 1px 29px 0px rgb(1 1 1 / 10%);"
            >
              <div
                class="text-center h-auto card-header"
              >
                <div
                  class="card-title h5"
                >
                  Welcome to Would You Rather App!
                </div>
                <div
                  class="card-subtitle h6"
                >
                  Please sign in to continue
                </div>
              </div>
              <div
                class="px-2 card-body"
              >
                <img
                  alt="would-you-rather"
                  class="card-img"
                  src="group.png"
                />
                <p
                  class="py-2 text-center card-text"
                  style="font-size: 1.5rem; font-weight: 600; color: rgb(16, 180, 103); margin-bottom: 0px;"
                >
                  Sign In
                </p>
                <form
                  class="px-2 d-flex align-items-center flex-column "
                >
                  <select
                    aria-label="select user"
                    class="form-select"
                    role="userSelect"
                  >
                    <option
                      value=""
                    >
                      Select User
                    </option>
                  </select>
                  <button
                    class="my-2 w-100 btn btn-primary"
                    disabled=""
                    style="font-size: 1.12rem; text-transform: none; font-weight: 600;"
                    type="button"
                  >
                    Sign In
                  </button>
                  <div
                    class="text-muted mt-2"
                    style="float: right;"
                  />
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
